test/playlist.test.js

```diff
@@ -0,0 +1,115 @@
+/* global describe, it, beforeEach */
+const Player = require('../src/js/video');
+const Subject = require('../src/js/playlist');
+const sinon = require('sinon/pkg/sinon');
+
+function createPlayer () {
+	const stub = sinon.createStubInstance(Player);
+	stub.opts = {};
+	stub.containerEl = document.createElement('div');
+	stub.videoEl = document.createElement('video');
+	stub.update.returns(Promise.resolve());
+
+	return stub;
+}
+
+describe('Playlist', () => {
+	let player;
+	const queue = ['foo', 'bar', 'baz', 'qux'];
+
+	beforeEach(() => {
+		player = createPlayer();
+	});
+
+	describe('constructor', () => {
+		it('can instantiate', () => {
+			const instance = new Subject({ player, queue: [] });
+
+			instance.should.be.an.instanceOf(Subject);
+
+			instance.opts.should.exist;
+			instance.opts.player.should.equal(player);
+		});
+
+		it('selects currently playing video from playlist', () => {
+			player.videoData = { id: 'bar' };
+
+			const instance = new Subject({ player, queue });
+
+			instance.currentIndex.should.equal(1);
+		});
+
+		it('starts the playlist if the current video does not match', () => {
+			const instance = new Subject({ player, queue });
+
+			instance.currentIndex.should.equal(0);
+			sinon.assert.calledOnce(player.update);
+		});
+
+		it('listens for the video to end to trigger the next in the queue', () => {
+			player.videoData = { id: 'bar' };
+
+			new Subject({ player, queue });
+
+			// no DOM so trigger this on the listener directly
+			player.containerEl.dispatchEvent(new CustomEvent('ended', { bubbles: false }));
```

Comment:
As it's a unit test, how we've decided to test this is by spying on addEventListener and making sure that it´s being set and that the handler is the correct one